example/src/test/scala/StepDefinitions.scala

```diff
@@ -1,21 +1,34 @@
-class StepDefinitions extends StepDefinitionsTrait {
+class StepDefinitions extends StepDefinitionsTrait with CustomTypes {
 
-  When("""^I add (\d+) and (\d+)$"""){ (arg1: Double, arg2: Double) =>
+  When("""^I add (\d+) and (\d+)$""") { (arg1: Double, arg2: Double) =>
     calc push arg1
     calc push arg2
     calc push "+"
   }
 
-  When("I sub (\\d+)" + " and " + "(\\d+)") {
-    (arg1: Double, arg2: Double) =>
-      calc push arg1
-      calc push arg2
-      calc push "-"
+  When("I sub (\\d+)" + " and " + "(\\d+)") { (arg1: Double, arg2: Double) =>
+    calc push arg1
+    calc push arg2
+    calc push "-"
   }
 
   When("I div " + (5 + 5) + " by " + (10 - 8)) {
     calc push 10.0
     calc push 2.0
     calc push "/"
   }
-}
\ No newline at end of file
+
+  When("I move at {speed} for {time}") { (arg1: Speed, arg2: Time) =>
```

Comment:
{speed} and {time} will be replaced by .* and .*

Reply:
But how will "10m/s" be translated into `Speed` ? Will it work?

Reply:
Yes, it uses the type extractor to do it. But if you have an overloaded funciton lets say.
speed-km-per-h and speed-m-per-s and you use km/h then intellij will suggest both and you have to select the right one yourself. Nor will it help you if you don't have any converter configured. 
So it helps with navigation but not with 'validation' of your code.

Reply:
Thank you, I think I get it now.